### PR TITLE
50144_admin_book_create

### PR DESCRIPTION
**Pull request by manazirahsan**
This is only "create" of Books' CRUD.
Nested attributes of BookImage will be implemented later.

### CODE REVIEW

**Comment by noor-framgia** [comment]
ok

File: app/views/admin/books/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<% provide(:title, t("books_page_title")) %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/new.html.erb]: ()

File: app/views/admin/books/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+<h3><%= t :books_new_add_new_book %></h3>
+<%= render "form" %>
```
Comment (abdullahallmehedi) [inline app/views/admin/books/new.html.erb]: :form

File: .byebug_history [hunk 3]
```diff
@@ -0,0 +1,7 @@
+continue
+params[:book]
```
Comment (dieunb) [inline .byebug_history]: checkout

File: app/controllers/admin/books_controller.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+class Admin::BooksController < ApplicationController
+  def index
+  end
+
+  def show
+  end
+
+  def new
+    @book = Book.new
```
Comment (dieunb) [inline app/controllers/admin/books_controller.rb]: cancancan

File: app/controllers/admin/books_controller.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  def create
+    @book = Book.new book_params
```
Comment (dieunb) [inline app/controllers/admin/books_controller.rb]: cancancan

File: app/views/admin/books/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<% provide(:title, t(:books_page_title)) %>
```
Comment (dieunb) [inline app/views/admin/books/new.html.erb]: remove `()`

File: config/routes.rb [hunk 7]
```diff
@@ -7,6 +7,11 @@
     get "contact"=> "static_pages#contact"
 
     resources :requests, only: [:new, :create, :index]
+    namespace :admin do
```
Comment (dieunb) [inline config/routes.rb]: admin root page

File: db/seeds.rb [hunk 8]
```diff
@@ -5,3 +5,8 @@
 #
 #   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
 #   Mayor.create(name: 'Emanuel', city: cities.first)
+Category.create name: "Science"
```
Comment (dieunb) [inline db/seeds.rb]: should use [faker](https://github.com/stympy/faker) and [fabricationgem](http://www.fabricationgem.org/) to fake data

File: app/views/admin/books/new.html.erb (hunk 6 above)
Comment (manazirahsan) [inline app/views/admin/books/new.html.erb]: it does not work without braces ()

File: app/controllers/admin/books_controller.rb (hunk 4 above)
Comment (manazirahsan) [inline app/controllers/admin/books_controller.rb]: CanCanCan will be added in later pull request.

File: app/controllers/admin/books_controller.rb (hunk 5 above)
Comment (manazirahsan) [inline app/controllers/admin/books_controller.rb]: CanCanCan will be added in later pull request.

File: config/routes.rb (hunk 7 above)
Comment (manazirahsan) [inline config/routes.rb]: Books' index view is not implemented yet.
It will be added later and admin root will be "admin::books#index" later.

Comment (dieunb) [inline config/routes.rb]: add now  `root "books#index"`

**Pull request merged**